cdap-explore-client/src/main/java/co/cask/cdap/explore/service/Explore.java

```diff
@@ -261,14 +261,26 @@ QueryHandle getTables(@Nullable String catalog, @Nullable String schemaPattern,
    * Get information about a Hive table.
    *
    * @param namespace name of the namespace the table belongs to.
-   * @param table table name for which to get the schema.
+   * @param table table name for which to get the info.
    * @return information about a table.
    * @throws ExploreException on any error getting the tables.
    */
   TableInfo getTableInfo(String namespace, String table)
     throws ExploreException, TableNotFoundException;
 
   /**
+   * Get information about a Hive table.
+   *
+   * @param namespace name of the namespace the table belongs to.
+   * @param database database name for which to get the info. If null, the database is derived from the namespace.
+   * @param table table name for which to get the info.
+   * @return information about a table.
+   * @throws ExploreException on any error getting the tables.
+   */
+  TableInfo getTableInfo(String namespace, @Nullable String database, String table)
+    throws ExploreException, TableNotFoundException;
```

Comment:
Should `TableNotFoundException` be listed as a thrown exception? I'm guessing there are numerous methods in this class that don't list this exception in the Javadocs.